src/hooks/useLocaleFormatter/useLocaleFormatter.ts

```diff
@@ -187,14 +187,19 @@ export const useLocaleFormatter = (args?: useLocaleFormatterArgs): NumberFormatt
         minimumFractionDigits,
         lessThanMin ? 6 : getFiatNumberFractionDigits(number),
       )
+      // Filter out undefined options caused by optional component props so they do not override the defaults
+      const filteredOptions = options
+        ? Object.fromEntries(Object.entries(options).filter(([_, value]) => value !== undefined))
+        : undefined
```

Comment:
Prefer `{}` over `undefined` as spreading `undefined` technically works but feels off

Reply:
Sure, although `undefined` is what is spread when options is not defined in the function call if I'm not mistaken 😄 
But you're right, I'll make the change, it's more readable anyways.